src/geometry/types.rs

```diff
@@ -119,8 +119,60 @@ impl Vector {
             dy: self.dy / magnitude,
         })
     }
+
+// Rectangle /////////////////////////////////////////////////////////////////
+pub struct Size{
+    pub width: f32,
+    pub heigh: f32,
+}
+
+pub struct Rectangle {
+    // the leftmost coordination system of the rectangle
+    pub origin: Point,
+    // the width and high of the rectangle
+    pub size: Size,
 }
 
+impl Rectangle{
+    pub fn new(point: Point, size: Size) -> Rectangle{
+        Rectangle{origin: point, size: size}
+    }
+
+    pub fn half_size(&mut self) {
+        self.size.width = self.size.width/2.0;
+        self.size.heigh = self.size.heigh/2.0;
+    }
+
+    pub fn quadrants(&self) -> Vec<Rectangle> {
+        let mut v:Vec<Rectangle> = Vec::new();
```

Comment:
I would suggest not using a `Vec` here: we are always going to return four `Rect` objects, so we can just return then in a tuple. This gives us the benefit of restructuring, like this:

`let (a, b, c, d) = some_rect.quadrants();`

Reply:
so the return type should be (Rect,Rect,Rect,Rect) ?

Reply:
Yes, but don't forget the spaces! Like this `(Rect, Rect, Rect, Rect)`